telemetry/mixpanel.go

```diff
@@ -42,20 +43,39 @@ func (m MixpanelTelemetryTracker) TrackEvent(eventContext EventContext, eventPro
 	// Combine our baseline props that we send for _ALL_ events with the passed in props from the event
 	trackProps := mergeMaps(baseProps, eventProps)
 
-	err := m.client.Track(m.runId, eventContext.EventName, trackProps)
-
+	request := map[string]interface{}{
+		"id":         m.runId,
+		"event":      eventContext.EventName,
+		"eventProps": trackProps,
+	}
+	jsonStr, err := json.Marshal(request)
+	if err != nil {
+		log.Println(err.Error())
+		return
+	}
+	req, err := http.NewRequest("POST", m.url, bytes.NewBuffer(jsonStr))
+	if err != nil {
+		log.Println(err.Error())
+		return
+	}
+	req.Header.Set("Content-Type", "application/json")
+	resp, err := m.client.Do(req)
+	if err != nil {
+		log.Println(err.Error())
+		return
+	}
```

Comment:
```suggestion
	resp, err := m.client.Post(m.url, "application/json", bytes.NewBuffer(jsonStr))
	if err != nil {
		log.Println(err.Error())
		return
	}
```
NIT: You can also use [http.Post](https://pkg.go.dev/net/http#Post), under the hood it does the same thing you are doing. `client` is `http.Client{}` so this should work.

https://github.com/golang/go/blob/master/src/net/http/client.go#L833-L858